src/app/api/event/[eventId]/routes.ts

```diff
@@ -0,0 +1,31 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { zObjectId } from '@/types/dataModel/base';
+import CMError, { CMErrorResponse, CMErrorType } from '@/utils/cmerror';
+import { zUpdateEventRequest } from '@/types/dataModel/event';
+import { updateEventAction } from '@/server/actions/Event';
+import { UpdateEventRequest } from '@/types/dataModel/event';
+
+export async function PUT(
+  req: NextRequest,
+  {
+    params,
+  }: { params: { eventId: string; eventUpdatesRequest: UpdateEventRequest } }
+) {
+  try {
+    const idValidationResult = zObjectId.safeParse(params.eventId);
+    if (!idValidationResult.success) {
+      return new CMError(CMErrorType.BadValue, 'Event Id').toNextResponse();
+    }
+
+    const data = await req.json();
+    const validationResult = zUpdateEventRequest.safeParse(data);
+    if (!validationResult.success) {
+      return new CMError(CMErrorType.BadValue, 'Event').toNextResponse();
+    }
+    await updateEventAction(params.eventId, params.eventUpdatesRequest);
```

Comment:
right here you will probably just use data instead of params.eventUpdateRequest.